qt/scientific_interfaces/Inelastic/Manipulation/InelasticDataManipulationElwinTab.cpp

```diff
@@ -417,33 +332,38 @@ void InelasticDataManipulationElwinTab::addDataToModel(MantidWidgets::IAddWorksp
 
 void InelasticDataManipulationElwinTab::updateTableFromModel() {
   m_view->clearDataTable();
-  for (auto domainIndex = FitDomainIndex{0}; domainIndex < m_dataModel->getNumberOfDomains(); domainIndex++) {
-    m_view->addTableEntry(static_cast<int>(domainIndex.value), m_dataModel->getWorkspace(domainIndex)->getName(),
-                          static_cast<int>(m_dataModel->getSpectrum(domainIndex)));
-  }
+  if (m_view->isRowCollapsed())
+    for (auto workspaceIndex = WorkspaceID{0}; workspaceIndex < m_dataModel->getNumberOfWorkspaces(); workspaceIndex++)
+      m_view->addTableEntry(static_cast<int>(workspaceIndex.value),
+                            m_dataModel->getWorkspace(workspaceIndex)->getName(),
+                            m_dataModel->getSpectra(workspaceIndex).getString());
+  else
+    for (auto domainIndex = FitDomainIndex{0}; domainIndex < m_dataModel->getNumberOfDomains(); domainIndex++)
+      m_view->addTableEntry(static_cast<int>(domainIndex.value), m_dataModel->getWorkspace(domainIndex)->getName(),
+                            std::to_string(m_dataModel->getSpectrum(domainIndex)));
 }
 
 void InelasticDataManipulationElwinTab::handleRemoveSelectedData() {
   auto selectedIndices = m_view->getSelectedData();
   std::sort(selectedIndices.begin(), selectedIndices.end());
   for (auto item = selectedIndices.end(); item != selectedIndices.begin();) {
     --item;
-    m_dataModel->removeDataByIndex(FitDomainIndex(item->row()));
+    m_view->isRowCollapsed() ? m_dataModel->removeWorkspace(WorkspaceID(item->row()))
+                             : m_dataModel->removeDataByIndex(FitDomainIndex(item->row()));
   }
   updateTableFromModel();
   updateAvailableSpectra();
 }
 
+void InelasticDataManipulationElwinTab::handleRowModeChanged() { updateTableFromModel(); }
+
 void InelasticDataManipulationElwinTab::updateAvailableSpectra() {
   auto spectra = m_dataModel->getSpectra(findWorkspaceID());
-  if (m_view->getCurrentInputIndex() == 1) {
-    if (spectra.isContinuous()) {
-      auto const minmax = spectra.getMinMax();
-      m_view->setAvailableSpectra(minmax.first, minmax.second);
-    } else {
-      m_view->setAvailableSpectra(spectra.begin(), spectra.end());
-    }
-  }
+  if (spectra.isContinuous()) {
+    auto const minmax = spectra.getMinMax();
+    m_view->setAvailableSpectra(minmax.first, minmax.second);
+  } else
+    m_view->setAvailableSpectra(spectra.begin(), spectra.end());
```

Comment:
I might be wrong here - but would doing 
```
m_view->setAvailableSpectra(spectra.begin(), spectra.end());
```
work in both cases, where the spectra is continuous as well as not continuous?

Reply:
It seems to be working.